packages/react-dev-utils/shouldIgnoreNodeModules.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+// While Create React App does not use polling for watching files by default,
+// polling can occur if CHOKIDAR_USEPOLLING environment variable is
+// set to a truthy value. Excessive polling can cause CPU overloads
+// on some systems (see https://github.com/facebookincubator/create-react-app/issues/293),
+// which is why we ignore node_modules if polling is enforced.
+let shouldIgnoreNodeModules = false;
+
+const envForceUsePolling = process.env.CHOKIDAR_USEPOLLING;
+
+if (envForceUsePolling) {
+  const usePollingLower = envForceUsePolling.toLowerCase();
+
+  // Chokidar rules https://github.com/paulmillr/chokidar/blob/master/index.js#L99
+  if (
+    usePollingLower === 'true' ||
+    usePollingLower === '1' ||
+    !!usePollingLower
+  ) {
+    shouldIgnoreNodeModules = true;
+  }
+}
+
+if (process.platform === 'darwin') {
+  // On OSX polling can also occur if chokidar fails to load the
+  // `fsevents` module. Check if the module is loaded:
+  const cacheKeys = Object.keys(require.cache);
+  const cachedFsEvents =
+    require.cache[cacheKeys.find(i => i.includes('/fsevents/fsevents.js'))];
+
+  if (cachedFsEvents && cachedFsEvents.loaded) {
+    // fsEvents is loaded, check if its WebpackDevServers
+    let parentIsWebpackDevServer = false;
+    let parent = cachedFsEvents.parent;
+
+    while (parent && !parentIsWebpackDevServer) {
+      if (parent.id.includes('webpack-dev-server')) {
```

Comment:
I'm not super happy about this in particular. In the future we'll migrate off WDS. Do we need to hardcode this at all?
Can you explain more about why this check is necessary?

Reply:
Like mentioned somewhere about, the check is _most likely_ not necessary, but anyway: consider case where ModuleX and WebpackDevServer both load Chokidar, and due to _something_ ModuleX properly loads `fsevents` but WDS fails to do so -> WDS polls eventhough it didn't successfully load the module.

That said,
 - I don't know if that's even possible 🤔 
 - Nothing else in the CRA stack currently depends on Chokidar, so the check is currently rather moot anyway.

I am totally okay with removing it if you are 🆗

Reply:
Is it WDS require-ing fsevents, or webpack? I would assume that webpack should be somewhere on the stack because WDS doesn't (afaik) do any watching manually, it just tells webpack to watch.

I'd be fine to change this line if it referenced just webpack.

Reply:
Sadly its [WDS that requires `chokidar`](https://github.com/webpack/webpack-dev-server/blob/00e8500b0853312be3cf369976509fbce2a4b7dc/lib/Server.js#L10) to which the `watchOptions` are passed in [WDS's own implementation of something similar to `watchpack`](https://github.com/webpack/webpack-dev-server/blob/00e8500b0853312be3cf369976509fbce2a4b7dc/lib/Server.js#L674-L697) which eventually requires [`fsevents`](https://github.com/paulmillr/chokidar/blob/34432b371269412a3ff007329d2982b5dd2e8430/lib/fsevents-handler.js#L7).